### PR TITLE
refactor(core): don't use extension macro for core js

### PR DESCRIPTION
**Pull request by nayeemrmn**
Seems like too much of a special case because `init_cbs()` needs to be called right after them. Removes the `Extension::is_core` stuff.

Blocked on #19615.

### CODE REVIEW

**Review by bartlomieju — COMMENTED** [review]
Blocked by https://github.com/denoland/deno/pull/19615

File: core/runtime/jsruntime.rs [hunk 1]
```diff
@@ -840,6 +841,23 @@ impl JsRuntime {
     let mut esm_entrypoints = vec![];
 
     futures::executor::block_on(async {
+      if self.init_mode == InitMode::New {
+        let js_files = include_js_files!(
+          core
+          "00_primordials.js",
+          "01_core.js",
+          "02_error.js",
```
Comment (bartlomieju) [inline core/runtime/jsruntime.rs:849]: Making this inline extension will not print `rerun-if-changed:` directive which will not rebuild core once these JS files change. Can you fix that?

**Review by bartlomieju — COMMENTED** [review]
Spotted one problem, but otherwise great cleanup!

**Comment by nayeemrmn** [comment]
Extracted the other small cleanups from #19519

**Review by bartlomieju — APPROVED** [review]
LGTM, thank you @nayeemrmn, it's much easier to land these cleanups in small chunks 👍

**Pull request merged**